common.eai

```diff
@@ -9753,37 +9756,43 @@ function AddHarassUnittype takes integer groupnum, integer harassnum, group g, g
   local unit u = null
   local real player_sum = 0
   set cg = SelectNumberOfId(cg, qty, old_id[uid])
+  set cg = SelectByUnitFree(cg, true)
+  set cg = SelectByUnitStandard(cg, true)  //Reduce usage IsUnitInGroup
   loop
     set u = FirstOfGroup(cg)
     exitwhen u == null
-    call GroupAddUnit(unit_harassing, u)
-    call RemoveGuardPosition(u)
     call GroupAddUnit(harasser,u)
     set player_sum = player_sum + GetUnitStrength(u)
     call GroupRemoveUnit(cg,u)
     call GroupRemoveUnit(g,u)
   endloop
   call DestroyGroup(cg)
   set cg = null
-  call SaveReal(additional_info, key, START_STRENGTH, player_sum)
+  call SaveReal(additional_info, key, START_STRENGTH_TEMP, player_sum)
   return harasser
 endfunction
 
 function StartHarass takes integer groupnum, integer harass_target, boolean avoid_towers, real strength_limit, real flee_percent, integer flee_number, real lx, real ly returns nothing
   local integer i = 0
   local integer t = 0
-  local group harasser = CreateGroup()
-  local group g = CreateGroup()
+  local group harasser = null
+  local group g = null
   local integer key = groupnum
   local real strength = 0
+  local real getstrength = 0
   local integer qty = 0
   local integer uid = 0
+  if HaveSavedHandle(additional_info, key, HARASS_GROUP) then
```

Comment:
I don't think we have any mechanism to clear this, so this will just permanently be stuck all game if you add this.

Reply:
At least not until the strategy changes.

Reply:
when the harass job ends, the API will be called to close, which can achieve the logical loop of this function. That's why I have this restriction. If you think the official API is not trustworthy, you can write your own state. However, I did not submit the changes to JOB together. Maybe I need to make some changes and submit the complete code together. In short, I think this logical problem is not significant

It's a bit strange here. Previously, I had a complete logic that ongoing harassment would not be restarted, but new units would be added. Later, when detecting a crash, it was removed. However, the code for temporary intensity calculation has been retained, but it's not very useful anymore

I'll see how to make some changes and replace some content later

Reply:
`call RemoveSavedHandle(additional_info, key, HARASS_GROUP)`
harass job ends use this , so maybe is clear mechanism

Reply:
change some thing , try look look

`if not check `HaveSavedHandle` , just continuously writing , groups may also experience excretion problems`